### PR TITLE
Referrals: Send pass UI

### PR DESCRIPTION
**Pull request by SergioEstevao**
| 📘 Part of: #2083  |  
|:---:|

Fixes # 

This PR implements the UI only for the Send Pass UI for Referrals.

> It still does not implement the background animated gradient on the Cards. That will be done a separate PR.

| 1 | 2 | 3 | 
| - | - | - |
| Simulator Screenshot - iPhone 15 Pro - 2024-09-04 at 16 54 06 | Simulator Screenshot - iPad mini (6th generation) - 2024-09-04 at 16 56 39 | Simulator Screenshot - iPad mini (6th generation) - 2024-09-04 at 16 56 24 | 

## To test

1. Start the app
2. Ensure that you have the Transcripts FF enabled and you logged with a Plus or Patron account
3. Go to Profile View Controller
4. Tap on the gift icon on the top left
5. Check if the new guest pass screen is show
6. Tap on the close button
7. Tap on the share guest pass button
8. Test on the iPad

## Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [x] I have considered adding unit tests for my changes.
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: podcasts/Referrals/ReferralCardView.swift [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+                    .foregroundColor(.white)
+                    .padding()
+
+            }
+            .overlay(alignment: .topTrailing) {
+                Image("plusGold")
+                    .renderingMode(.template)
+                    .resizable()
+                    .aspectRatio(contentMode: .fill)
+                    .frame(width: 12, height: 12)
```
Comment (bjtitus) [inline podcasts/Referrals/ReferralCardView.swift]: Should this be in Constants? I'm not really sure I like it any better having so many Constants and just jumping between lines but I noticed the `defaultCardSize` defined as a constant in another file.l

**Review by bjtitus — APPROVED** [review]
Looked good in testing on both iPhone and iPad.

Left a small note but I'm not really sure about the whole Constants thing in SwiftUI TBH.

File: podcasts/Referrals/ReferralCardView.swift (hunk 1 above)
Comment (SergioEstevao) [reply podcasts/Referrals/ReferralCardView.swift]: Addressed here: https://github.com/Automattic/pocket-casts-ios/pull/2120/commits/2b0689fc534d8c9a0d21052559fb503fb7ef368d

For the record I'm also a bit split when to use the enum Constants or not, I like to use them when the value is used in multiplace spaces or to make clear the intention of the value.

**Pull request merged**